tcms/testruns/views.py

```diff
@@ -271,36 +272,71 @@ def _walk_executions(test_executions):
                comments_subtotal.get(case_run.pk, 0))
 
 
-@permission_required('testruns.change_testrun')
-def edit(request, run_id):
+@method_decorator(permission_required('testruns.change_testrun'), name='dispatch')
+class TestRunEditView(FormView):
     """Edit test plan view"""
 
-    try:
-        test_run = TestRun.objects.select_related().get(run_id=run_id)
-    except ObjectDoesNotExist:
-        raise Http404
+    template_name = 'testruns/mutable.html'
+    form_class = BaseRunForm
+
+    def form_invalid(self, form):
+        test_run = self._get_test_run_or_fail(self.kwargs['run_id'])
+
+        context_data = {
+            'test_run': test_run,
+            'test_plan': test_run.plan,
+            'form': self._get_base_form(test_run),
+            'errors': form.errors
+        }
 
-    # If the form is submitted
-    if request.method == "POST":
-        form = BaseRunForm(request.POST)
+        return render(self.request, self.template_name, context_data)
+
+    def form_valid(self, form):
+        run_id = self.kwargs['run_id']
+        test_run = self._get_test_run_or_fail(run_id)
+        test_run.summary = form.cleaned_data['summary']
+        # Permission hack
+        if test_run.manager == self.request.user or test_run.plan.author == self.request.user:
+            test_run.manager = form.cleaned_data['manager']
```

Comment:
What do we hack here exactly?

Reply:
Note: this should probably be updated only if changed.

Reply:
Here I only replace access to the request object to be via self keyword.

Reply:
I meant the comment above the if statement. It looks like this could be just an old stale comment which doesn't match the existing logic so remove it.